lib/new_relic/agent/transaction.rb

```diff
@@ -337,10 +337,8 @@ def set_default_transaction_name(name, category)
 
       def set_overriding_transaction_name(name, category)
```

Comment:
Maybe we could define a new shared `set_transaction_name` which accepts an `override` boolean parameter, defaulting to `false`. Then both `set_default_transaction_name` and `set_overriding_transaction_name` can call that shared method with the latter setting `override` to `true`.